src/transformers/models/fastvit/modeling_fastvit.py

```diff
@@ -0,0 +1,1103 @@
+# coding=utf-8
+# Copyright 2023 Apple, Ross Wightman, The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch FastViT model."""
+
+
+import collections.abc
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...modeling_outputs import BaseModelOutput, BaseModelOutputWithPooling, ImageClassifierOutput
+from ...modeling_utils import PreTrainedModel
+from ...pytorch_utils import find_pruneable_heads_and_indices, prune_linear_layer
+from ...utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+)
+from .configuration_fastvit import FastViTConfig
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "FastViTConfig"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "JorgeAV/fastvit_t8"
+_EXPECTED_OUTPUT_SHAPE = [1, 384, 8, 8]
+
+# Image classification docstring
+_IMAGE_CLASS_CHECKPOINT = "JorgeAV/fastvit_t8"
+_IMAGE_CLASS_EXPECTED_OUTPUT = "tabby, tabby cat"
+
+# TODO: Add more new models there are atleast 5 more
+FASTVIT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "JorgeAV/fastvit_t8"
+    # See all FastViT models at https://huggingface.co/models?filter=fastvit
+]
+
+
+class FastViTEmbeddings(nn.Module):
+    """
+    Construct the patch embeddings. Optionally, also the mask token.
+    """
+
+    def __init__(self, config: FastViTConfig) -> None:
+        super().__init__()
+
+        self.patch_embeddings = FastViTPatchEmbeddings(config)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+    def forward(self, pixel_values: torch.Tensor) -> torch.Tensor:
+        embeddings = self.patch_embeddings(pixel_values)
+        embeddings = self.dropout(embeddings)
+
+        return embeddings
+
+
+class FastViTPatchEmbeddings(nn.Module):
```

Comment:
classes that are used in other classes should be defined before. Same for functions